packages/scandipwa/src/component/CurrencySwitcher/CurrencySwitcher.component.js

```diff
@@ -47,13 +57,34 @@ export class CurrencySwitcher extends PureComponent {
         return availableCurrencies.some((e) => e.id === currency) ? currency : currentCurrencyCode;
     }
 
+    onCurrencyChange(currencyCode) {
+        const { handleCurrencySelect } = this.props;
+        const { pathname = '' } = location;
+
+        if (pathname.match(CART_URL) || pathname.match(appendWithStoreCode(CHECKOUT_URL))) {
+            this.setState({ isPageReloading: true });
+            document.documentElement.style.setProperty('--page-overflow', 'hidden');
+            document.documentElement.style.setProperty('--cookie-z-index', '-1');
```

Comment:
1. Please use `CSS.setVariable` from 'Util/CSS' to set the CSS custom property.
2. You change the CSS variable name globally for all elements (from document root). And the previous value will not be restored on component unmount. So either:
- The initial value isn't necessary, since everything on the page looks good with the new ones, and you can set these values as initial
- You have a bug
If you need to put custom CSS variable values to a specific component, please apply the new values to that specific component. You can use React refs for this purpose.
3. These values aren't dynamic. You put static string values. Please move the CSS variable value override to the CSS class and apply this class conditionally.

Reply:
Thought I don't need to reset these variables, 
The idea here is to set these variables for a second (while the request gets sent) till the page reloads.

Didn't account for if the request returns an error since it wasn't accounted for in the config.dispatcher.

But, in the next commit I added a catch method that resets the variables and the state.

Reply:
WOW, didn't expect that the currency switcher value change would reload the SPA.